code/modules/admin/topic/topic_teleports.dm

```diff
@@ -155,3 +155,57 @@
 			else
 				to_chat(owner, SPAN_ALERT("Mobs choice error. Aborting."))
 				return
+
+		if("teleport_corpses")
+			if(alert(owner, "[GLOB.dead_mob_list.len] corpses are marked for teleportation. Pressing \"TELEPORT\" will teleport them to your location at the moment of pressing button.", "Confirmation", "Teleport", "Cancel") == "Cancel")
+				return
+			for(var/mob/M in GLOB.dead_mob_list)
+				if(!M)
+					continue
+				M.on_mob_jump()
+				M.forceMove(get_turf(owner.mob))
+			message_staff(WRAP_STAFF_LOG(owner.mob, "mass-teleported [GLOB.dead_mob_list.len] corpses to themselves in [get_area(owner.mob)] ([owner.mob.x],[owner.mob.y],[owner.mob.z])."), owner.mob.x, owner.mob.y, owner.mob.z)
+		
+		if("teleport_items_by_type")
+			var/item = input(owner,"What item?", "Item Fetcher","") as text|null
+			if(!item)
+				return
+
+			var/list/types = typesof(/atom)
+			var/list/matches = new()
+
+			//Figure out which object they might be trying to fetch
+			for(var/path in types)
+				if(findtext("[path]", item))
+					matches += path
+
+			if(matches.len==0)
+				return
+
+			var/chosen
+			if(matches.len==1)
+				chosen = matches[1]
+			else
+				//If we have multiple options, let them select which one they meant
+				chosen = tgui_input_list(usr, "Select an atom type", "Spawn Atom", matches)
+
+			if(!chosen)
+				return
+
+			//Find all items in the world
+			var/list/targets = list()
+			for(var/obj/item/M in GLOB.item_list)
+				if(istype(M, chosen))
+					to_world("YES")
+					targets += M
+
+			if(alert(owner, "[targets.len] items are marked for teleportation. Pressing \"TELEPORT\" will teleport them to your location at the moment of pressing button.", "Confirmation", "Teleport", "Cancel") == "Cancel")
```

Comment:
Could add a check for 0 found

Reply:
Added a check both here and for corpses.